src/main/java/it/gov/pagopa/payment/connector/rest/reward/RewardCalculatorConnectorImpl.java

```diff
@@ -0,0 +1,41 @@
+package it.gov.pagopa.payment.connector.rest.reward;
+
+import feign.FeignException;
+import it.gov.pagopa.common.performancelogger.PerformanceLog;
+import it.gov.pagopa.payment.connector.rest.reward.dto.AuthPaymentRequestDTO;
+import it.gov.pagopa.payment.connector.rest.reward.dto.AuthPaymentResponseDTO;
+import it.gov.pagopa.payment.exception.ClientExceptionNoBody;
+import it.gov.pagopa.payment.exception.ClientExceptionWithBody;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+@Service
+public class RewardCalculatorConnectorImpl implements RewardCalculatorConnector {
+
+  private final RewardCalculatorRestClient restClient;
+
+  public RewardCalculatorConnectorImpl(RewardCalculatorRestClient restClient) {
+    this.restClient = restClient;
+  }
+
+  @Override
+  @PerformanceLog("QR_CODE_AUTHORIZE_TRANSACTION_REWARD_CALCULATOR")
+  public AuthPaymentResponseDTO authorizePayment(String initiativeId, AuthPaymentRequestDTO body) {
+    AuthPaymentResponseDTO responseDTO = new AuthPaymentResponseDTO();
+    try {
+      responseDTO = restClient.authorizePayment(initiativeId, body);
+    } catch (FeignException e) {
+      switch (e.status()) {
+        case 409 -> {
+          return responseDTO;
+        }
+        case 429 ->
+            throw new ClientExceptionWithBody(HttpStatus.TOO_MANY_REQUESTS, "REWARD CALCULATOR",
+                "Too many request in the microservice reward-calculator");
+        default -> throw new ClientExceptionNoBody(HttpStatus.INTERNAL_SERVER_ERROR,
+            "An error occurred in the microservice reward-calculator");
+      }
+    }
+    return responseDTO;
```

Comment:
map here the returned model, in order to not let the @Service to know about connector's models

At row 24 don't create an emtpy object, but use this variable to store the object to return, then at row 39 invoke the mapper